hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -1938,6 +1938,14 @@ public boolean isInStandbyState() {
         HAServiceState.OBSERVER == haContext.getState().getServiceState();
   }
 
+  public boolean isInObserverState() {
+    if (haContext == null || haContext.getState() == null) {
+      return haEnabled;
```

Comment:
Seems like this was probably copied from `isInStandbyState()`? But I don't think it's right. If we can't find a state, we assume `STANDBY` state. If we assume `STANDBY` state because a valid state could not be found, then `isInObserverState()` should be false. So I think we should just `return false` here.

Reply:
I agree with you.

Reply:
> Seems like this was probably copied from `isInStandbyState()`? But I don't think it's right. If we can't find a state, we assume `STANDBY` state. If we assume `STANDBY` state because a valid state could not be found, then `isInObserverState()` should be false. So I think we should just `return false` here.

I agree with you. Thanks.